rfcs/text/0010-email.md

```diff
@@ -1,68 +1,44 @@
--->
+This RFC proposes a new top-level field set to facilitate email use cases, `email.*`. The `email.*` field set adds fields for the sender, recipient, message header fields, and other attributes of an email message typically seen logs produced by mail transfer agent (MTA) and email gateway applications.
 
 ## Fields
 
 <!--
 Stage 1: Describe at a high level how this change affects fields. Which fieldsets will be impacted? How many fields overall? Are we primarily adding fields, removing fields, or changing existing fields? The goal here is to understand the fundamental technical implications and likely extent of these changes. ~2-5 sentences.
 -->
 
-Email specific fields:
+### Email specific fields
 
 | field | type | description |
 | --- | --- | --- |
-| `email.bcc.addresses` | `wildcard[]` | Addresses of Bcc's |
-| `email.cc.addresses` | `wildcard[]` | Addresses of Cc's |
-| `email.attachments_count` | long | A field outside the flattened structure to control how many attachments are included in the email |
-| `email.attachments` | flattened | A flattened field for anything related to attachments. This allows objects being stored with all information for each file when you have multiple attachments |
-| `email.direction` | keyword | Direction of the message based on the sending and receving domains |
-| `email.sender.address` | wildcard | Senders email address |
-| `email.sender.domain` | wildcard | Domain of the sender |
-| `email.sender.top_level_domain` | keyword | Top level domain of the sender |
-| `email.sender.registered_domain` | wildcard | Registered domain of the sender |
-| `email.sender.subdomain` | keyword | Subdomain of the sender |
-| `email.message_id` | keyword | Internet message ID of the message |
-| `email.reply_to.address` | wildcard | Reply-to address |
-| `email.return_path.address` | wildcard | The return address for the message |
-| `email.size` | long | Total size of the message, in bytes, including attachments |
-| `email.subject` | wildcard | Subject of the message |
-| `email.recipients.addresses` | `keyword[]` | Recipient addresses |
-| `email.domains` | `keyword[]` | domains related to the email |
-
-
-Other ECS fields used together with email usecases:
-| field | description |
-| --- | --- |
-| `event.duration` | The duration related to the email event. Could be the total duration in Quarantine, how long the email took to send from source to destination etc |
-| `event.start` | When the email event started
-| `event.end` | When the email event ended
-| `process.name` | When the event is related to a server or client. Does not take MTA into account which is part of a ongoing discussion |
-| `network.protocol` | Type of email protocol used |
-| `tls.*` | Used for TLS related information for the connection to for example a SMTP server over TLS |
-
-
+| `email.from` | keyword | Stores the `from` email address |
```

Comment:
It should be clarified that this is referring to the RFC5322.From or "Header From" address specifically

There should be another field for RFC5321.MailFrom "Envelope From" and/or Return-Path as RFC5321.MailFrom and RFC5322.From often differ

Reply:
Thanks @ericwbentley 

So, something like this:
| - | - | - |
| - | - | - |
| `email.from` | keyword | Stores the `from` email address provided in the "From" originator field |

Reference: https://datatracker.ietf.org/doc/html/rfc5322#section-3.6.2

Reply:
I'd reference the RFC in the description and add a second field for the envelope from address since they should be distinct from each other.

```suggestion
| `email.from` | keyword | Stores the header `from` email address (RFC5322.From) |
| `email.envelope_from` | keyword | Stores the envelope `from` email address (RFC5321.MailFrom) |
```

Reply:
@ericwbentley Makes sense to reference RFC 5322 - thanks for suggesting the distinction.

Like I put in the [`Concerns` section](https://github.com/P1llus/ecs/blob/master/rfcs/text/0010-email.md#email-messages-vs-protocols), right now, this proposal focuses on the IMF from RFC5322. However, I like the idea of also introducing an `smtp` field set later that focuses on details of the protocol (and other email protocols, if helpful).

I see `smtp.*` fields pairing well with `email.*`, but I want to avoid increasing the scope here too much. Open to feedback, though.